alibi_detect/cd/pytorch/context_aware.py

```diff
@@ -265,5 +265,5 @@ def _sigma_median_diag(x: torch.Tensor, y: torch.Tensor, dist: torch.Tensor) ->
     The computed bandwidth, `sigma`.
     """
     n_median = np.prod(dist.shape) // 2
-    sigma = (.5 * dist.flatten().sort().values[n_median].unsqueeze(dim=-1)) ** .5
+    sigma = (.5 * dist.flatten().sort().values[int(n_median)].unsqueeze(dim=-1)) ** .5
```

Comment:
**Note**: @jklaise if when you come to review you're wondering why the error is only apparent now, I suspect it is because `numpy>=1.23.0` is now installable due to the change in core deps. See https://github.com/SeldonIO/alibi-detect/pull/539#issuecomment-1167205579.